kernel/src/storage_permissions.rs

```diff
@@ -20,21 +23,26 @@ use core::cmp;
 /// we may want to revamp this interface.
 #[derive(Clone, Copy)]
 pub struct StoragePermissions {
-    // How many entries in the `read_permissions` slice are valid, starting at
-    // index 0.
+    /// How many entries in the `read_permissions` slice are valid, starting at
+    /// index 0.
     read_count: usize,
-    // Up to eight 32 bit identifiers of storage items the process has read
-    // access to.
+    /// Up to eight 32 bit identifiers of storage items the process has read
+    /// access to.
     read_permissions: [u32; 8],
-    // How many entries in the `write_permissions` slice are valid, starting at
-    // index 0.
+    /// How many entries in the `write_permissions` slice are valid, starting at
+    /// index 0.
     write_count: usize,
-    // Up to eight 32 bit identifiers of storage items the process has write
-    // (update) access to.
+    /// Up to eight 32 bit identifiers of storage items the process has write
+    /// (update) access to.
     write_permissions: [u32; 8],
-    // The identifier for this storage user when creating new objects. If `None`
-    // there is no `write_id` for these permissions.
-    write_id: Option<u32>,
+    /// The identifier for this storage user when creating new objects. If
+    /// `None` there is no `write_id` for these permissions.
+    write_id: Option<NonZeroU32>,
+    /// If `kerneluser` is true, this permission grants access to all objects
+    /// stored stored with `write_id` 0. New items created with `kerneluser ==
+    /// true` will use the specified ID if  `write_id.is_some()`, otherwise new
+    /// items will be created with the reserved ID (i.e., 0).
+    kerneluser: bool,
```

Comment:
Do we need this? Isn't the `write_id` Option enough

Reply:
`write_id.is_none()` means these permissions do not allow writes at all, so we wouldn't be able to differentiate "no writes" from "kernel writing".